### PR TITLE
feat: upgraded to node v18, added .nvmrc and updated workflows

### PR DESCRIPTION
**Pull request by BilalQamar95**
## Ticket
[Upgrade Node Js from 16 to 18](https://github.com/openedx/frontend-app-gradebook/issues/318)

## Description
- Added support for node v18, updated .nvmrc and workflows. 
- Updated `frontend-platform` & `frontend-build` versions

## edx-internal PR
https://github.com/edx/edx-internal/pull/8338
This needs to be merged together.

**Developer Checklist**
- [ ] Test suites passing
- [ ] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

File: src/data/services/segment/utils.test.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const courseId = 'a-fake-course-id';
 const category = 'AFakeCategory';
 describe('segment service utils', () => {
   beforeAll(() => {
-    global.window = Object.create(window);
+    global.window ??= Object.create(window);
```
Comment (muselesscreator) [inline src/data/services/segment/utils.test.js:12]: What is this?

Comment (muselesscreator) [inline src/data/services/segment/utils.test.js:12]: Why this statement?  The goal here is explicitly to replace the global window

Comment (BilalQamar95) [reply src/data/services/segment/utils.test.js]: That was to resolve `TypeError: Cannot redefine property: window`. This appeared to be a [bug](https://github.com/nodejs/node/issues/47563) which was fixed by [#46615](https://github.com/nodejs/node/pull/46615) and back-ported to node v18.15 however it is still failing for node v18.16. For now I have pinned the node version to v18.15.

**Review by muselesscreator — APPROVED** [review]
lgtm

**Pull request merged**